facebookads/test/integration.py

```diff
@@ -46,35 +46,70 @@ class FacebookAdsTestCase(unittest.TestCase):
     TEST_ID = str(int(time.time()) % 1000)
     TEST_IMAGE_PATH = os.path.join(os.path.dirname(__file__), 'test.png')
 
-    @classmethod
-    def new_test_ad_creative(cls):
-        creative = objects.AdCreative(
-            parent_id=cls.TEST_ACCOUNT.get_id_assured()
-        )
+    def setUp(self):
+        super(FacebookAdsTestCase, self).setUp()
+        self.remote_objects = []
+
+    def tearDown(self):
+        self.delete_remote_objects()
+        super(FacebookAdsTestCase, self).tearDown()
+
+    def delete_remote_objects(self):
+        """
+        Delete accumulated remote objects.
+        """
+        for o in reversed(self.remote_objects):
+            if o.Field.id in o and o.get_id() is not None:
+                try:
+                    o.remote_delete()
+                except Exception:
+                    if isinstance(o, objects.AdImage):
+                        # AdImages are often reused automatically since the
+                        # hash is unique. They can't be deleted until all the
+                        # references are cleaned up.
+                        pass
+                    else:
+                        # Error out the test if an object can't be deleted,
+                        # since this probably indicates a coding issue in the
+                        # tests.
+                        raise
+
+    def delete_me(self, obj):
```

Comment:
Let's rename this to something that tells a better story: `delete_in_teardown` or something